docs/tuning_fledge.rst

```diff
@@ -99,3 +104,149 @@ The Fledge core contains a scheduler that is used for running periodic tasks, th
+
+Configuring Storage Plugins
+###########################
+
+The storage plugins to use can be selected in the *Advanced* section of the *Configuration* page. Select the *Storage* category from the category tree display and the following will be displayed.
+
++------------------+
+| |storage_config| |
++------------------+
+
+  - **Storage Plugin**: The name of the storage plugin to use. This will be used to store the configuration data and must be one of the supported storage plugins. This can not be the *sqlitememory* plugin as that plugin does not support the storage of configuration.
+
+  - **Reading Plugin**: The name of the storage plugin that will be used to store the readings data. If left blank then the *Storage Plugin* above will be used to store both configuration and readings.
+
+  - **Database threads**: Increase the number of threads used within the storage service to manage the database activity. This is not the number of threads that can be used to read or write the database and increasing this will not improve the throughput of the data.
+
+  - **Manage Storage**: This is used when an external storage application, such as the Postgres database is used that requires separate initialization. If this external process is not run by default setting this to true will cause Fledge to start the storage process. Normally this is not required as Postgres should be run as a system service and SQLite does not require it.
+
+  - **Service Port**: Normally the storage service will dynamically create a service port that will be used by the storage service. Setting this to a value other than 0 will cause a fixed port to be used. This can be useful when developing a new storage plugin or to allow access to a non-fledge application to the storage layer. This should only be changed with extreme caution.
+
+  - **Management Port**: Normally the storage service will dynamically create a management port that will be used by the storage service. Setting this to a value other than 0 will cause a fixed port to be used. This can be useful when developing a new storage plugin.
+
+Changing will be saved once the *save* button is pressed. Fledge uses a mechanism whereby this data is not only saved in the configuration database, but also cached to a file called *storage.json* in the *etc* directory of the data directory. This is required such that Fledge can find the configuration database during the boot process. If the configuration becomes corrupt for some reason simply removing this file and restarting Fledge will cause the default configuration to be restored.
```

Comment:
> data directory.

 if explicit data is set then FLEDGE_DATA directory else FLEDGE_ROOT/data directory.

Also we have an outstanding issue when we directly edit the storage.json file, GUI does not come up with right value of storage engine as configured in storage.json. See FOGL-6023